process/block/metablock.go

```diff
@@ -2139,30 +2139,64 @@ func (mp *metaProcessor) verifyTotalAccumulatedFeesInEpoch(metaHdr *block.MetaBl
 	return nil
 }
 
-func (mp *metaProcessor) computeAccumulatedFeesInEpoch(metaHdr data.MetaHeaderHandler) (*big.Int, *big.Int, error) {
-	currentlyAccumulatedFeesInEpoch := big.NewInt(0)
-	currentDevFeesInEpoch := big.NewInt(0)
+func (mp *metaProcessor) getCurrentlyAccumulatedFees(metaHdr data.MetaHeaderHandler) (*big.Int, *big.Int, error) {
+	if metaHdr.IsHeaderV3() {
+		lastExecResult, err := common.GetLastBaseExecutionResultHandler(metaHdr)
+		if err != nil {
+			return nil, nil, err
+		}
 
-	lastHdr := mp.blockChain.GetCurrentBlockHeader()
-	if !check.IfNil(lastHdr) {
-		lastMeta, ok := lastHdr.(*block.MetaBlock)
+		lastMetaExecResult, ok := lastExecResult.(data.BaseMetaExecutionResultHandler)
 		if !ok {
 			return nil, nil, process.ErrWrongTypeAssertion
 		}
 
-		if !lastHdr.IsStartOfEpochBlock() {
-			currentlyAccumulatedFeesInEpoch = big.NewInt(0).Set(lastMeta.AccumulatedFeesInEpoch)
-			currentDevFeesInEpoch = big.NewInt(0).Set(lastMeta.DevFeesInEpoch)
-		}
+		return lastMetaExecResult.GetAccumulatedFeesInEpoch(), lastMetaExecResult.GetDevFeesInEpoch(), nil
+	}
+
+	lastHdr := mp.blockChain.GetCurrentBlockHeader()
+	if check.IfNil(lastHdr) {
+		return big.NewInt(0), big.NewInt(0), nil
+	}
+
+	lastMeta, ok := lastHdr.(*block.MetaBlock)
+	if !ok {
+		return nil, nil, process.ErrWrongTypeAssertion
+	}
+
+	if lastHdr.IsStartOfEpochBlock() {
+		return big.NewInt(0), big.NewInt(0), nil
 	}
+	currentlyAccumulatedFeesInEpoch := big.NewInt(0).Set(lastMeta.AccumulatedFeesInEpoch)
+	currentDevFeesInEpoch := big.NewInt(0).Set(lastMeta.DevFeesInEpoch)
+
+	return currentlyAccumulatedFeesInEpoch, currentDevFeesInEpoch, nil
+}
+
+func (mp *metaProcessor) getHeaderAccumulatedFees(metaHdr data.MetaHeaderHandler) (*big.Int, *big.Int) {
+	if metaHdr.IsHeaderV3() {
+		gasAndFees := mp.getGasAndFees()
+		return gasAndFees.GetAccumulatedFees(), gasAndFees.GetDeveloperFees()
+	}
+
+	return metaHdr.GetAccumulatedFees(), metaHdr.GetDeveloperFees()
+}
+
+func (mp *metaProcessor) computeAccumulatedFeesInEpoch(metaHdr data.MetaHeaderHandler) (*big.Int, *big.Int, error) {
+	currentlyAccumulatedFeesInEpoch, currentDevFeesInEpoch, err := mp.getCurrentlyAccumulatedFees(metaHdr)
```

Comment:
we also need handling for the epoch change proposed block -> when executing that block, the accumulated fees for the previous epoch are used to compute the rewards and distributed or burned, and the accumulatedFees and accumulatedDevFees are reset.

I think we miss the reset.

Reply:
The reset should be done if this is the epoch change proposed block.

Reply:
updated getCurrentlyAccumulatedFees to return 0 in case of epoch change proposed block.. missed it